test/getJwk.spec.js

```diff
@@ -0,0 +1,159 @@
+'use strict'
+
+const nock = require('nock')
+const t = require('tap')
+
+const buildGetJwks = require('../src/get-jwks')
+
+t.beforeEach((done, t) => {
+  nock.disableNetConnect()
+  done()
+})
+
+t.afterEach((done, t) => {
+  nock.cleanAll()
+  nock.enableNetConnect()
+  done()
+})
+
+t.test('getJwk should return an error if the request fails', async t => {
+  nock('https://localhost/').get('/.well-known/jwks.json').reply(500, { msg: 'no good' })
+  try {
+    const getJwks = buildGetJwks()
+    await getJwks.getJwk({ domain: 'https://localhost/', alg: 'ALG', kid: 'SOME_KEY' })
+  } catch (e) {
+    t.equal(e.message, 'Internal Server Error')
+    t.same(e.body, { msg: 'no good' })
+  }
```

Comment:
what happens if no error is thrown in this test?

Reply:
I see!  It's not something I thought about, thank you!